web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,864 @@
+        },
+        /**
+         * @override
+         */
+        getFocusableElement: function () {
+            return this.$target || $();
+        },
+        /**
+         * Do not re-render this field if it was the origin of the onchange call.
+         *
+         * @override
+         */
+        reset: function (record, event) {
+            var value = this.value;
+            this._reset(record, event);
+            if (this.nodeOptions.wrapper) {
+                value = this._wrap(value);
+            }
+            value = this._textToHtml(value);
+            if (!event || event.target !== this) {
+                if (this.mode === 'edit') {
+                    this.wysiwyg.setValue(value);
+                } else {
+                    this.$content.html(value);
+                }
+            }
+            return Promise.resolve();
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        _getValue: function () {
+            if (this.markdownEditor){
+                var value = this._getHtmlValue(this.markdownEditor[0].value);
+            } else {
+                var value = this.$target.val();
+            }
+
+            if (this.nodeOptions.wrapper) {
+                return this._unWrap(value);
+            }
+            return value;
+        },
+        /**
+         * Create the wysiwyg instance with the target (this.$target)
+         * then add the editable content (this.$content).
+         *
+         * @private
+         * @returns {$.Promise}
+         */
+        _createWysiwygIntance: function () {
+            var self = this;
+            this.wysiwyg = new Wysiwyg(this, this._getWysiwygOptions());
+            //coming from markdown we need to inject in new wysiwyg new value.
+            //This is done, cleanly and elegantly in getValue
+            this.wysiwyg.__extraAssetsForIframe = this.__extraAssetsForIframe || [];
+            // by default this is synchronous because the assets are already loaded in willStart
+            // but it can be async in the case of options such as iframe, snippets...
+            return this.wysiwyg.attachTo(this.$target).then(function () {
+                self.$content = self.wysiwyg.$editor.closest('body, odoo-wysiwyg-container');
+                self.isRendered = false;
+                self._isDirty = true;
+                self._onLoadWysiwyg();
+                self.$el.find('div#radio_container').after("<div id='md_infoarea'></div>");
+                var infoarea = self.$el.find('div#md_infoarea');
+                // look for a comment with the tag in it , if you find it remove the
+                // comment.
+                if (self._html_has_md_source_tag()) {
```

Comment:
So I think the main-main issue is that IF you find this tag, which should be a `script` tag with the markdown content in it, then:

- DONT call `self.sw_html_to_md`, because this is lossy, 
- INSTEAD just load the markdown directly from the contents of this tag, which is not lossy, as it's been saved there on last edit and not touched since. (if it HAS been touched, then the tag wont be there, because we should have a write hook in Python or some other hook that makes sure that if anything OTHER than the markdown widget has touched it, the markdown tag is removed).

Reply:
correct, but the main problem is that <script></script>   content  is destroyed on editor save.
If we don;t save, the content is extracted correctly from the tag and put in markdown.

This code is launched on load, and it never will be used, because currently we cant save the <script> so it will never find it.  Actually   self.sw_html_to_md()   does not exist.  the name is:    self.sw_html_to_mk() 


  self.sw_html_to_md()  does already exactly what you ask:  it calls "getMarkdownValue" and getMarkdownValue has been modified to search for the tag and if it's there use the inner text.
So that will be lossless.

The "quick conversion" via tag is reliable and works, if we don;t save.

Reply:
function name corrected.